packages/strapi-plugin-content-manager/controllers/collection-types.js

```diff
@@ -0,0 +1,226 @@
+
+    await wrapBadRequest(async () => {
+      const updatedEntity = await entityManager.update(entity, sanitizeFn(body), model);
+
+      ctx.body = permissionChecker.sanitizeOutput(updatedEntity);
+    })();
+  },
+
+  async delete(ctx) {
+    const { userAbility } = ctx.state;
+    const { id, model } = ctx.params;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.delete()) {
+      return ctx.forbidden();
+    }
+
+    const entity = await entityManager.findOneWithCreatorRoles(id, model);
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    if (permissionChecker.cannot.delete(entity)) {
+      return ctx.forbidden();
+    }
+
+    const result = await entityManager.delete(entity, model);
+
+    ctx.body = permissionChecker.sanitizeOutput(result);
+  },
+
+  async publish(ctx) {
+    const { userAbility } = ctx.state;
+    const { id, model } = ctx.params;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.publish()) {
+      return ctx.forbidden();
+    }
+
+    const entity = await entityManager.findOneWithCreatorRoles(id, model);
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    if (permissionChecker.cannot.publish(entity)) {
+      return ctx.forbidden();
+    }
+
+    const result = await entityManager.publish(entity, model);
+
+    ctx.body = permissionChecker.sanitizeOutput(result);
+  },
+
+  async unpublish(ctx) {
+    const { userAbility } = ctx.state;
+    const { id, model } = ctx.params;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.unpublish()) {
+      return ctx.forbidden();
+    }
+
+    const entity = await entityManager.findOneWithCreatorRoles(id, model);
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    if (permissionChecker.cannot.unpublish(entity)) {
+      return ctx.forbidden();
+    }
+
+    const result = await entityManager.unpublish(entity, model);
+
+    ctx.body = permissionChecker.sanitizeOutput(result);
+  },
+
+  async bulkdDelete(ctx) {
+    const { userAbility } = ctx.state;
+    const { model } = ctx.params;
+    const { query, body } = ctx.request;
+    const { ids } = body;
+
+    const entityManager = getService('entity-manager');
+    const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+    if (permissionChecker.cannot.delete()) {
+      return ctx.forbidden();
+    }
+
+    const permissionQuery = permissionChecker.buildPermissionQuery(query);
+
+    const idsWhereClause = { [`id_in`]: ids };
+    const params = {
+      _limit: 100,
```

Comment:
Why do we hardcode this limit here?

Reply:
I only see one valid reason: to avoid too much processing as we are finding the entires then deleting them one by one. I just reused the existing query tbh. I will actually refactor to limit the length of the ids array instead :)